src/components/PeoplePage/PeoplePage.tsx

```diff
@@ -0,0 +1,70 @@
+import { useEffect, useState } from 'react';
+import { Loader } from '../Loader';
+import { Person } from '../../types';
+import { getPeople } from '../../api';
+import { PeopleTable } from '../PeopleTable';
+import { PeopleContext } from '../../peopleContext';
+import { PeopleFilters } from '../PeopleFilters';
+import { useSearchParams } from 'react-router-dom';
+import { filterPeople } from '../../utils/filterPeople';
+
+export const PeoplePage = () => {
+  const [people, setPeople] = useState<Person[]>([]);
+  const [filteredPeople, setFilteredPeople] = useState<Person[]>(people);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+
+  const [searchParams] = useSearchParams();
+
+  useEffect(() => {
+    setError(false);
+    setLoading(true);
+
+    getPeople()
+      .then(res => {
+        setPeople(res);
+        setFilteredPeople(() => res);
+      })
+      .catch(() => setError(true))
+      .finally(() => setLoading(false));
+  }, []);
+
+  useEffect(() => {
+    setFilteredPeople(() => filterPeople(people, searchParams));
```

Comment:
You're filtering people every time the `people` array or the `searchParams` change. Depending on the size of your `people` array this could potentially be an expensive operation. Consider adding a button for the user to click when they want to perform the search, so that you're not performing it on every change.